themes/default/content/blog/managing-github-with-pulumi/index.md

```diff
@@ -0,0 +1,599 @@
+authors: ["guinevere-saenger"]
+tags: ["go", "migration", "packages"]
+---
+
+Ah, GitHub. The home of all developers. The place where we share code. The world’s most awkward social media site. The secret LinkedIn for techies. The tool we use for company org structure, work planning, code ownership, and permissions…
+
+Wait.
+
+That’s quite a lot.
+
+GitHub is good at many things, but a full-on organization management tool it is not.
+
+Have you ever needed your manager to manually enable admin permission on a repo for you?  
+Or have you needed to page the CEO to add you to a team, because your manager was out that day?  
+Have you ever wondered who is on what team? Or which team owns a repo?  
+What if you change teams, or a team changes names? A reorg happens, and the “platform-integrations” team is no more, but we still need to call it that on GitHub because it is the team with all the repository accesses?
+
+When I joined Pulumi in 2021, all of the above happened to me within my first few weeks.
+
+We at Pulumi wanted to reduce this kind of management friction, and we decided to solve it the Pulumi way: with declarative infrastructure using the [Pulumi GitHub provider](https://github.com/pulumi/pulumi-github).
+
+<!--more-->
+
+## The Goal
+
+First, we set a few ground rules. We decided that administrative privileges over Pulumi's github repositories should flow as follows:
+
+1. An employee is part of one or more GitHub teams.
+2. Teams have appropriately scoped permissions over each repository they own.
+3. Any Pulumi employee should be able to request code repository access in an official way.
+
+Additionally, any current team membership and repository access should not be disrupted during development of this tool.
+
+## Getting Started
+
+Create a new Pulumi program:
+
+```bash
+$ pulumi new go
+This command will walk you through creating a new Pulumi project.
+
+Enter a value or leave blank to accept the (default), and press <ENTER>.
+Press ^C at any time to quit.
+
+project name: (team-management)
+project description: (A minimal Go Pulumi program)
+Created project 'team-management'
+
+Please enter your desired stack name.
+To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
+stack name: (dev) pulumi/prod
+Created stack 'prod'
+
+Installing dependencies...
+
+go: downloading github.com/pulumi/pulumi/sdk/v3 v3.26.1
+Finished installing dependencies
+
+Your new project is ready to go! ✨
+
+To perform an initial deployment, run 'pulumi up'
+
+$ ls
+Pulumi.yaml go.mod  go.sum  main.go
+```
+
+We now have a pulumi project yaml, and the beginnings of a small Go program all set up.
+
+According to the [provider configuration documentation](https://www.pulumi.com/registry/packages/github/installation-configuration/), we need to add a properly scoped token, as well as set "pulumi" as our GitHub organization owner.
```

Comment:
I'm pretty sure the main docs build will fail if we don't use `relref`s here (and anywhere linking within pulumi.com in the post) unless something has recently changed. Here's what that would be:

```suggestion
According to the [provider configuration documentation]({{< relref "/registry/packages/github/installation-configuration" >}}), we need to add a properly scoped token, as well as set "pulumi" as our GitHub organization owner.
```

Reply:
is there a different docs build on `master`? because this PR went green!

Reply:
oh didnt realize there was a thread will delete my other comment, but we now do a full docs build and broken links will be reported on hugo prs. since https://github.com/pulumi/pulumi-hugo/pull/1106

Reply:
that being said, we prefer rel refs so we can actually do these checks and better be able to read error messages when there are issues

Reply:
Ahh, so the relrefs don't break the docs build any longer. I knew it was running a full build; I had figured it just somehow wasn't catching those as they always broke mine. No worries!

Reply:
well now im worried im making this up. i guess @cnunciato should confirm.

Reply:
Yep, @susanev's right on both counts -- we do run a full build to ensure the PR won't fail when it's pulled into pulumi/docs (though we don't use it for previews, as it'd tack on many minutes to each PR job), and we do prefer relative links -- ideally `{{< relref >}}`s, as they prevent broken links -- over fully-qualified ones.

There are a few reasons for this (which we can elaborate on in the TBD content guide), but the main one is that hard-coded URLs create problems on other hosts (e.g., in local development, in PR previews, other environments, etc.). So it's generally considered a best practice to stick to root-relative paths as a rule.